test/cpp/api/modules.cpp

```diff
@@ -103,6 +104,59 @@ TEST_F(ModulesTest, Conv3d) {
   ASSERT_TRUE(model->weight.grad().numel() == 3 * 2 * 3 * 3 * 3);
 }
 
+TEST_F(ModulesTest, MaxPool1d) {
+  MaxPool1d model(MaxPool1dOptions(3).stride(2));
+  auto x = torch::randn({1, 1, 5}, torch::requires_grad());
+  auto y = model(x);
+  torch::Tensor s = y.sum();
+
+  s.backward();
+  std::cout << y.sizes() << std::endl;
+  ASSERT_EQ(y.ndimension(), 3);
+  ASSERT_EQ(s.ndimension(), 0);
+  ASSERT_EQ(y.sizes(), torch::IntArrayRef({1, 1, 2}));
```

Comment:
Besides checking `ndimension()` and `sizes()`, can we also check that the values of `y` are what we expected? (we might need to set value of `x` in a specific way in order to test this)